src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultActionModelBuilder.cs

```diff
@@ -256,6 +257,15 @@ private bool IsIDisposableMethod(MethodInfo methodInfo, TypeInfo typeInfo)
             AddRange(actionModel.ActionConstraints, attributes.OfType<IActionConstraintMetadata>());
             AddRange(actionModel.Filters, attributes.OfType<IFilter>());
 
+            foreach (var authorizeAttribute in attributes.OfType<AuthorizeAttribute>())
+            {
+                actionModel.Filters.Add(new AuthorizeFilter
+                {
+                    Policy = authorizeAttribute.Policy,
+                    Roles = authorizeAttribute.Roles?.Split(','),
+                });
+            }
```

Comment:
We need to see the same in `ControllerModelBuilder`. This code only looks at attributes on the method.